### ISSUE TITLE
Can't generate .csproj for player projects

### ISSUE THREAD

**Issue by Merichbier** [microsoft/vscode-dotnettools#763]
Type: <b>Bug</b>

Currently it seems impossible to toggle the generation of player projects file. This is usefull when developing for platform where we need to write native code, like for HoloLens 2.

In Visual Studio we have the possibility to switch the code analysis between project and player (editor / platform) which allows compilation checks for the platform.
![image](https://github.com/microsoft/vscode-dotnettools/assets/9365869/4f56414a-3a12-4d4a-8f86-b16b8c29a62e)
![image](https://github.com/microsoft/vscode-dotnettools/assets/9365869/35380714-88bb-4c51-af7e-0361560fc053)

Reproduction steps :
- Go to Unity Preferences
- Try to toggle Player projects under "Generate .csproj files for:"
![image](https://github.com/microsoft/vscode-dotnettools/assets/9365869/c974cfc2-203d-46c7-9fbf-144265306584)

Actual:
- Can't toggle the option

Expected:
- Toggle is active

Extension version: 0.9.3
VS Code version: Code 1.84.2 (1a5daa3a0231a0fbba4f14db7ec463cf99d7768e, 2023-11-09T10:51:52.184Z)
OS version: Windows_NT x64 10.0.22621
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i9-12900H (20 x 2918)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.68GB (9.13GB free)|
|Process Argv|C:\\git-repos\\some-repo --crash-reporter-id fc47f6da-83c7-43c9-b339-54ad6c91ce78|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263:30880072
vsclangdc:30486549
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
282f8724:30602487
f6dab269:30613381
showlangstatbar:30737416
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
welcomedialog:30887143
pythonnosmt12:30797651
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
dsvsc013:30795093
dsvsc014:30804076
dsvsc015:30845448
pythontestfixt:30871694
pythonregdiag2:30871582
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
2e7ec940:30885897
pythontbext0:30879054
accentitlementsc:30887149
dsvsc016:30886110
dsvsc017:30886112
dsvsc018:30886114
aa_t_chat:30882232

```

</details>

**Comment by dibarbet** [comment]
@jbevain is this expected to be supported?  How do these player projects work?  Is the .cs file shared between multiple projects - or is there a separate .cs file for each project?

If these projects required shared files - then assuming the project generation happens, we'd still need to implement https://github.com/dotnet/vscode-csharp/issues/5788 to allow you to switch between different contexts.

**Comment by jbevain** [comment]
Hi @dibarbet, at the moment we're not supporting this because there's no way to switch between different contexts :)

Basically today we generate project files to represent what Unity is compiling for the Unity Editor.

Having Player projects allows you in VS to be able to select the project context to represent what Unity is compiling when building a Player for a target platform, so you can get a different editor experience based on ifdef or references for instance.

**Comment by dibarbet** [comment]
> Basically today we generate project files to represent what Unity is compiling for the Unity Editor.
> 
> Having Player projects allows you in VS to be able to select the project context to represent what Unity is compiling when building a Player for a target platform, so you can get a different editor experience based on ifdef or references for instance.

How does this work in VS - is the project generated with a linked file reference to the same .cs file?  If so we can mark this as blocked on Roslyn support for project contexts.

**Comment by jbevain** [comment]
Yes, there are 2 csprojs with a different set of defines or references that are referencing the same .cs file.

**Comment by dibarbet** [comment]
Got it, blocked on https://github.com/dotnet/vscode-csharp/issues/5788 then

**Comment by Merichbier** [comment]
[https://github.com/dotnet/vscode-csharp/issues/5788](https://github.com/dotnet/vscode-csharp/issues/5788) has been fix in the preview release. Can we hope this issue will be investigated ?

**Comment by jbevain** [comment]
@Merichbier thanks for the heads-up! That's great news, I missed it. Happy to test it and see about enabling Player projects generation for VSCode/C# DevKit.

**Comment by arthurboyer** [comment]
That would be a game changer in my opinion for multi-platform development with VSCode ! I hope this feature will come soon.

**Comment by jbevain** [comment]
Player project generation for VS Code/C# DevKit is coming in the “Visual Studio Editor” package version 2.0.28 in Unity. We don't have an ETA for when this will land but we'll close this when it's available.

**Comment by jbevain** [comment]
The "Visual Studio Editor" package 2.0.28 has finally been released by Unity. You should be able to enable Player projects now and use the project selector to change the IntelliSense context. Thank you!
